PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/table/ImportAppSettingsDialog.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (c) Microsoft Corporation
+ * <p/>
+ * All rights reserved.
+ * <p/>
+ * MIT License
+ * <p/>
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+ * documentation files (the "Software"), to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and
+ * to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ * <p/>
+ * The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+ * the Software.
+ * <p/>
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.microsoft.intellij.runner.functions.component.table;
+
+import com.intellij.icons.AllIcons;
+import com.intellij.ui.ListCellRendererWrapper;
+import com.intellij.ui.PopupMenuListenerAdapter;
+import com.intellij.ui.ToolbarDecorator;
+import com.microsoft.azure.management.appservice.FunctionApp;
+import com.microsoft.azuretools.core.mvp.model.ResourceEx;
+import com.microsoft.intellij.helpers.UIHelperImpl;
+import com.microsoft.intellij.runner.functions.AzureFunctionsConstants;
+
+import javax.swing.JButton;
+import javax.swing.JCheckBox;
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JDialog;
+import javax.swing.JLabel;
+import javax.swing.JList;
+import javax.swing.JPanel;
+import javax.swing.KeyStroke;
+import javax.swing.event.PopupMenuEvent;
+import java.awt.Dimension;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyEvent;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+public class ImportAppSettingsDialog extends JDialog implements ImportAppSettingsView {
+    private static final String LOCAL_SETTINGS_JSON = "local.settings.json";
+    private static final String TITLE_IMPORT_APP_SETTINGS = "Import App Settings";
+    private static final String LOADING_TEXT = "Loading...";
+    private static final String EMPTY_TEXT = "Empty";
+    private JPanel contentPane;
```

Comment:
shall we make a base dialog class with buttonOK , buttonCancel, contentPanel?

Reply:
contentPane -> contentPanel

Reply:
Nice idea, will try this later